src/ui/public/visualize/visualize.js

```diff
@@ -155,7 +159,17 @@ uiModules
           return searchSource.onResults().then(onResults);
         }).catch(notify.fatal);
 
-        searchSource.onError(notify.error).catch(notify.fatal);
+        searchSource.onError(e => {
+          if (isTermSizeZeroError(e)) {
```

Comment:
I like this idea, but it gets a little tricky given the implicit way our notification error messages are created. Let's address this as part of https://github.com/elastic/kibana/issues/8194